Vegetarian-Food-Finder/Controller/SearchViewController.swift

```diff
@@ -6,46 +6,67 @@
 //
 import UIKit
 
-final class SearchViewController: UIViewController, UITableViewDataSource, UITableViewDelegate, UISearchBarDelegate {
+final class SearchViewController: UIViewController, UITableViewDataSource, UITableViewDelegate, UISearchBarDelegate, UISearchResultsUpdating {
     private let searchBarCell = "SearchBarCell"
     private let segueFromSearchToDetails = "SegueFromSearchToDetails"
     @IBOutlet var tableView: UITableView!
     let searchController = UISearchController()
-    var restaurantsList: [Business] = []
+    var restaurantList: [Business] = []
+    var filteredRestaurantsList: [Business] = []
```

Comment:
You don't need to store the filtered restaurants every time the user makes a search. You can make the filter at the time you are making the presentation on screen